components/submit-expense/SubmitExpenseFlow.tsx

```diff
@@ -159,184 +58,47 @@ export function SubmitExpenseFlow(props: SubmitExpenseFlowProps) {
-          })),
-          longDescription: null,
-          privateMessage: null,
-          tags: values.tags,
-          tax: values.hasTax
-            ? [
-                {
-                  rate: values.tax.rate,
-                  type: formOptions.taxType,
-                  idNumber: values.tax.idNumber,
-                },
-              ]
-            : null,
-        };
-
-        if (formOptions.expense?.id && !startOptions.duplicateExpense) {
-          const editInput: EditExpenseFromDashboardMutationVariables['expenseEditInput'] = {
-            ...expenseInput,
-            id: formOptions.expense.id,
-            payee:
-              formOptions.expense?.status === ExpenseStatus.DRAFT && !formOptions.payee?.slug
-                ? formOptions.expense?.draft?.payee
-                : {
-                    slug: formOptions.payee?.slug,
-                  },
-          };
-          result = await editExpense({
-            variables: {
-              expenseEditInput: editInput,
-              draftKey: props.draftKey,
-            },
-          });
+  const onSuccess = React.useCallback(
+    (result, type: 'edit' | 'new' | 'invite') => {
+      setSubmittedExpenseId(result.data.expense.legacyId);
+      switch (type) {
+        case 'edit': {
           toast({
             variant: 'success',
             title: <FormattedMessage defaultMessage="Expense edited" id="yTblGN" />,
             message: LoggedInUser ? <Survey hasParentTitle surveyKey={SURVEY_KEY.EXPENSE_SUBMITTED_NEW_FLOW} /> : null,
             duration: 20000,
           });
-        } else if (
-          formOptions.payee?.type === CollectiveType.VENDOR ||
-          formOptions.payoutProfiles.some(p => p.slug === values.payeeSlug)
-        ) {
-          result = await createExpense({
-            variables: {
-              account: {
-                slug: formOptions.account?.slug,
-              },
-              expenseCreateInput: expenseInput,
-              ...(values.recurrenceFrequency !== RecurrenceFrequencyOption.NONE
-                ? {
-                    recurring: {
-                      interval: values.recurrenceFrequency as unknown as RecurringExpenseInterval,
-                      endsAt: dayjs(values.recurrenceEndAt).toDate(),
-                    },
-                  }
-                : {}),
-              privateComment:
-                formOptions.payoutMethod?.type === PayoutMethodType.BANK_ACCOUNT &&
-                formOptions.payoutMethod?.data?.accountHolderName !== formOptions.payee?.legalName
-                  ? values.payoutMethodNameDiscrepancyReason
-                  : null,
-            },
-          });
-
+          break;
+        }
+        case 'invite': {
           toast({
             variant: 'success',
-            title: <FormattedMessage id="Expense.Submitted" defaultMessage="Expense submitted" />,
+            title: <FormattedMessage defaultMessage="Expense invite sent" id="Fhue1N" />,
```

Comment:
Should we customize the messages based on Expense vs. Grant?

Reply:
Done in a different part of the code: https://github.com/opencollective/opencollective-frontend/pull/11084/commits/3b2c982835f44360bb572216da6976855514da70